### PR TITLE
Turb flow monitor

### PR DESCRIPTION
**Pull request by Rozie100**
The current monitor does not march over all components of the velocity field.

### CODE REVIEW

File: src/monitors/turbFlowStats.cpp [hunk 1]
```diff
@@ -112,12 +112,13 @@ PetscErrorCode ablate::monitors::TurbFlowStats::MonitorTurbFlowStats(TS ts, Pets
                     monitorPt[monitorFields[FieldPlacements::densitySum].offset] += densLoc;
                     monitorPt[monitorFields[FieldPlacements::densityDtSum].offset] += densLoc * dt;
 
-                    // Set the offset
-                    PetscInt offset = monitorFields[FieldPlacements::fieldsStart + f].offset;
+                    const auto& field = monitor->GetSolver()->GetSubDomain().GetField(monitor->fieldNames[f]);
 
                     // March over each field component
-                    for (int p = 0; p < monitorFields[f].numberComponents; p++) {
-                        offset += SectionLabels::END * p;
```
Comment (mmcgurn) [inline src/monitors/turbFlowStats.cpp:120]: Was this only broken when dim > 2? This change makes sense to me but it took a while to figure out the math.  Can you add a comment that explains the offset and how the components are arranged?

**Review by mmcgurn — APPROVED** [review]
Thanks for catching this!  I added a comment about adding some comments.

File: src/monitors/turbFlowStats.cpp (hunk 1 above)
Comment (Rozie100) [reply src/monitors/turbFlowStats.cpp:120]: Both 3D and 2D actually. The integration test, which is 2D gives garbage for the second velocity.

Comment (Rozie100) [reply src/monitors/turbFlowStats.cpp:120]: Tried to explain it. let me know if its is confusing

**Pull request merged**